internal/db/subscriptions.go

```diff
@@ -12,20 +12,29 @@ import (
 	"gorm.io/gorm/clause"
 )
 
-// QuotasFromPlan generates a set of quotas from the plan quota defaults in a plan. This function assumes that the
-// given plan already contains the plan quota defaults.
+// QuotasFromPlan generates a set of quotas from the plan quota defaults in a plan.
 func QuotasFromPlan(plan *model.Plan, periods int32) []model.Quota {
-	result := make([]model.Quota, len(plan.PlanQuotaDefaults))
-	for i, quotaDefault := range plan.PlanQuotaDefaults {
+
+	// Get the active plan quota defaults from the plan.
+	pqds := plan.GetDefaultQuotaValues()
+
+	// Build the array of quotas.
+	result := make([]model.Quota, len(pqds))
+
+	// Populate the quotas.
+	currentIndex := 0
+	for _, quotaDefault := range pqds {
 		quotaValue := quotaDefault.QuotaValue
 		if quotaDefault.ResourceType.Consumable {
 			quotaValue *= float64(periods)
 		}
-		result[i] = model.Quota{
+		result[currentIndex] = model.Quota{
 			Quota:          quotaValue,
 			ResourceTypeID: quotaDefault.ResourceTypeID,
 		}
+		currentIndex++
```

Comment:
I'm missing if there's a benefit to managing the `currentIndex` outside of the loop, since it appears to unconditionally increment it at the end of the loop iteration. It seems like this could be done more cleanly with `for currentIndex, quotaDefault := range pqds` (more similarly to the old code)

Reply:
The reason in this case is because the plan quota defaults are now in a map to ensure that we only have one plan quota default for each resource type.

I considered making another call to the database to get the active plan quota defaults, which would turn the data structure back into an array, but since we already have everything we need in memory, finding the correct plan quota defaults in code seemed to be the better option just to avoid another round trip to the database. I'm still on the fence about this decision, though. If enough plan quota defaults are added then the time saved by avoiding a round trip to the database might eventually become negligible (or be eliminated entirely).

Reply:
Ah! That makes sense. I had failed to notice it was a map, and thus wouldn't have indices to use.

Reply:
To be honest, I'd forgotten about it too and noticed it when I started to update the code. 😄